File: pyround1/387.字符串中的第一个唯一字符.py
#
# @lc app=leetcode.cn id=387 lang=python
#
# [387] 字符串中的第一个唯一字符
#

# @lc code=start
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        min_index = len(s)

        for c in "abcdefghijklmnopqrstuvwxyz":
            i  = s.find(c)
            if i != -1 and i == s.rfind(c):
                min_index = min(min_index, i)
        return min_index if min_index != len(s) else -1
            
# @lc code=end

File: 387.字符串中的第一个唯一字符.cpp
/*
 * @lc app=leetcode.cn id=387 lang=cpp
 *
 * [387] 字符串中的第一个唯一字符
 */

// @lc code=start
class Solution {
public:
    int firstUniqChar(string s) {
        if(s.empty())
            return -1;
        unordered_map<char, int> res;
        for(auto& ch : s)
            res[ch] += 1;
        for(int i= 0; i < s.size(); i++){
            if(res[s[i]] == 1)
                return i;
        }
        return -1;

        //-------------------
        for(int i = 0; i < s.size(); i++)
            if(s.find(s[i]) == s.rfind(s[i]))
                return i;
        return -1;
    }
};
// @lc code=end

